components/src/preact/lineageFilter/fetchLineageAutocompleteList.ts

```diff
@@ -1,20 +1,116 @@
+import { fetchLineageDefinition } from '../../lapisApi/lapisApi';
 import { FetchAggregatedOperator } from '../../operator/FetchAggregatedOperator';
 import type { LapisFilter } from '../../types';
 
 export async function fetchLineageAutocompleteList({
-    lapis,
-    field,
+    lapisUrl,
+    lapisField,
+    lapisFilter,
     signal,
+}: {
+    lapisUrl: string;
+    lapisField: string;
+    lapisFilter?: LapisFilter;
+    signal?: AbortSignal;
+}) {
+    const [countsByLineage, lineageTree] = await Promise.all([
+        getCountsByLineage({
+            lapisUrl,
+            lapisField,
+            lapisFilter,
+            signal,
+        }),
+        getLineageTree({ lapisUrl, lapisField, signal }),
+    ]);
+
+    return Array.from(lineageTree.keys())
+        .map((lineage) => {
+            return [
+                {
+                    lineage,
+                    count: countsByLineage.get(lineage) ?? 0,
+                },
+                {
+                    lineage: `${lineage}*`,
+                    count: getCountsIncludingSublineages(lineage, lineageTree, countsByLineage),
+                },
+            ];
+        })
+        .sort((a, b) => a[0].lineage.localeCompare(b[0].lineage))
+        .flat();
+}
+
+async function getCountsByLineage({
+    lapisUrl,
+    lapisField,
     lapisFilter,
+    signal,
 }: {
-    lapis: string;
-    field: string;
+    lapisUrl: string;
+    lapisField: string;
     lapisFilter?: LapisFilter;
     signal?: AbortSignal;
 }) {
-    const fetchAggregatedOperator = new FetchAggregatedOperator<Record<string, string>>(lapisFilter ?? {}, [field]);
+    const fetchAggregatedOperator = new FetchAggregatedOperator<Record<string, string>>(lapisFilter ?? {}, [
+        lapisField,
+    ]);
+
+    const countsByLineageArray = (await fetchAggregatedOperator.evaluate(lapisUrl, signal)).content;
+    return countsByLineageArray.reduce(
+        (acc, value) => acc.set(value[lapisField], value.count),
+        new Map<string, number>(),
+    );
```

Comment:
I think this does the same:
```suggestion
    return new Map<string, number>(countsByLineageArray.map((value) => [value[lapisField], value.count]));
```

Reply:
Good thinking.